client-app/src/desktop/common/Wrapper.js

```diff
@@ -20,10 +54,70 @@ class Wrapper extends Component {
                     marginBottom: 10,
                     omit: !description
                 }),
-                children
+                children,
+                this.renderSourceCodePanel()
             ],
             ...rest
         });
     }
+    
+    renderSourceCodePanel() {
+        const {dockContainerModel} = this;
+        const {links, link} = this.props;
+        
+        if (links || link) {
```

Comment:
We could invert the condition here and do an early return of null if there are no links (e.g. `if (!links) return null;`). This allows us to avoid having the bulk of the function in conditional nested params.

It's purely a stylistic choice, but one that I think helps make intentions clearer. There's a nice discussion here on the pros and cons of both styles: https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Reply:
Oohf.  Thanks for linking the discussion.  I've once had to make something with a lot of error handling, and the `if(somethingInvalid) return null` pattern was a lifesaver.  Much easier to ready than nested if and else's.  

I can see what you mean with inverting the condition -- it allows us to take the bulk of code out of the conditional.  While I don't usually think in this pattern, I agree it's probably good for reading and maintaining later.. 

Thanks again!